File: backend/app/schemas/batch.py
from pydantic import BaseModel
from typing import List

class BatchPredictionItem(BaseModel):
    customer_id: str
    churn_probability: float
    risk_level: str # "High", "Medium", "Low"

class BatchPredictionResponse(BaseModel):
    total_processed: int
    average_probability: float
    high_risk_count: int
    medium_risk_count: int
    low_risk_count: int
    predictions: List[BatchPredictionItem]

File: frontend/src/api/batch.ts
import { fetchWithAuth } from "./client";

export type BatchPredictionItem = {
  customer_id: string;
  churn_probability: number;
  risk_level: string;
};

export type BatchPredictionResponse = {
  total_processed: number;
  average_probability: number;
  high_risk_count: number;
  medium_risk_count: number;
  low_risk_count: number;
  predictions: BatchPredictionItem[];
};

export async function predictBatchCSV(file: File): Promise<BatchPredictionResponse> {
  const formData = new FormData();
  formData.append("file", file);

  const res = await fetchWithAuth(`/predict/batch/csv`, {
    method: "POST",
    body: formData, 
  }); 

  if (!res.ok) {
    const error = await res.json();
    throw new Error(error.detail || "Failed to process batch CSV");
  }
  return res.json();
}
